openpdf/src/main/java/com/lowagie/text/pdf/PdfStamperImp.java

```diff
@@ -896,11 +997,52 @@ else if (objReal instanceof PdfStream) {
                             }
                         }
                     }
-                    if (app != null) {
+                    if (app != null && objReal!=null) {
                         Rectangle box = PdfReader.getNormalizedRectangle(merged.getAsArray(PdfName.RECT));
                         PdfContentByte cb = getOverContent(page);
                         cb.setLiteral("Q ");
+
+                        if(transformNeeded) {
+                            AffineTransform transform = new AffineTransform();
+                            double x = box.getLeft();
+                            double y = box.getBottom();
+                            
+                            if (rotation != 0 && rotation % 90 == 0 && rotation % 270 != 0) {
+                                x += box.getWidth();
+                            }
+                            if (rotation != 0 && (rotation % 180 == 0 || rotation % 270 == 0)) {
```

Comment:
To test `rotation % 270 == 0` makes no sense, it is true for 270°, 540° (equivalent to 180°), 810° (equivalent to 90°), 1080° (equivalent to 0°), ...
I assume you mean something like `rotation % 360 == 180 || rotation % 360 == 270`.

Reply:
cp. above